### ISSUE TITLE
the chunked API does not appear in the readr function document

### ISSUE THREAD

**Issue by wehopkins** [tidyverse/readr#981]
I have readr 1.3.1 installed; none of the read_*_chunked or callback functions are in the generated function document for readr. I know of them only through a reference to them in Stack Overflow.

From looking at issue #76  (and #610) , it appears that they won't do what I want - allow actual incremental access to files to support very large file processing - but I assume they are of some use.

**Comment by jimhester** [comment]
This was intentional, we weren't sure how final the API was going to be so they were mostly experimental.

Can you describe more what you are trying to do and why you don't think the chunked API would work.

**Comment by wehopkins** [comment]
I want an efficient interface for incrementally reading in data from a very large text file (ten gigabytes), enrich each increment, and append into another file (or to a database). This avoids having to load the entire file into memory (pushes the limit on even my 128 GB memory server) and process the whole thing in place. The approach of using "skip" and "n_max" to chunk with read_csv reads from the front of the file for each chunk. And comments on related issues lead me to believe that the chunk API does the same.

The alternative I plan to evaluate is the LaF package.

**Comment by jimhester** [comment]
The chunked interface does not read from the beginning of the file for each chunk, you must be misinterpreting the comments.

**Comment by wehopkins** [comment]
Thanks. I'll have to try out the chunked API to read through the large files. Perhaps a brief vignette would help users better understand what the chunked functions do.

**Comment by RichardJActon** [comment]
Are there any plans to link up chunked reading with the future API - it would be really useful to be able to specify a callback function that operated on chunks of a file in parallel. For my purposes using something like this in conjunction with future.batchtools to read and process file chunks as batch jobs on an HPC would be great.
